src/common/github-flavored-markdown.js

```diff
@@ -104,24 +104,27 @@ class GithubFlavoredMarkdown {
     return [...linkSet];
   }
 
-  redactCdnResources(markdown, redactionCallback) {
-    if (!markdown) {
-      return markdown;
-    }
+  _redactUrls(markdown, redactionCallback) {
+    return typeof markdown === 'string'
+      ? markdown
+        .replace(imageInsideOfLinkPattern, (_match, g1, g2, g3, g4, g5) => `${g1}${redactionCallback(g2)}${g3}${redactionCallback(g4)}${g5}`)
+        .replace(simpleLinkOrImagePattern, (_match, g1, g2, g3) => `${g1}${redactionCallback(g2)}${g3}`)
+        .replace(autoLinkPattern, (_match, g1, g2, g3) => `${g1}${redactionCallback(g2)}${g3}`)
+      : markdown;
+  }
 
-    const redact = url => {
+  redactCdnResources(markdown, redactionCallback) {
+    return this._redactUrls(markdown, url => {
       const isRedactableUrl = isPortableCdnUrl(url) && !!getCdnPath({ url });
-      if (isRedactableUrl) {
-        return redactionCallback(url);
-      }
-
-      return url;
-    };
+      return isRedactableUrl ? redactionCallback(url) : url;
+    });
+  }
 
-    return markdown
-      .replace(imageInsideOfLinkPattern, (_match, g1, g2, g3, g4, g5) => `${g1}${redact(g2)}${g3}${redact(g4)}${g5}`)
-      .replace(simpleLinkOrImagePattern, (_match, g1, g2, g3) => `${g1}${redact(g2)}${g3}`)
-      .replace(autoLinkPattern, (_match, g1, g2, g3) => `${g1}${redact(g2)}${g3}`);
+  makeCdnResourcesPortable(markdown, cdnRootUrl) {
+    return this._redactUrls(markdown, url => {
+      const isInternalResourceUrl = isCdnUrl({ url, cdnRootUrl });
+      return isInternalResourceUrl ? getPortableUrl({ url, cdnRootUrl }) : url;
```

Comment:
Here we could even simplify to `return this._redactUrls(markdown, url => getPortableUrl({ url, cdnRootUrl }));` because `getPortableUrl` will simply return the url as is if there's no need to alter it.

Reply:
Cool, nice simplification! Done.